### ISSUE TITLE
Resolve deprecations and upgrade to gradle 8

### ISSUE THREAD

**Issue by clemens-tolboom** [molgenis/molgenis-service-armadillo#543]
We are using v 7.6.3

```zsh
./gradlew  --warning-mode all clean build docker assemble
...
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6.3/userguide/command_line_interface.html#sec:command_line_warnings
```

```
./gradlew --warning-mode all
```
gives 3 warnings in total. Make sure all tasks are touched

```
./gradlew  --warning-mode all clean build docker assemble
```

## Upgrade

```sh
./gradlew wrapper --gradle-version 8.0
```

**Comment by erikzwart** [comment]
Installed gradle 8.4 (Ubuntu 22.04) but did not change behaviour.

Second try was to change the distributionUrl in the file ./gradle/wrapper/gradle-wrapper.properties from:
distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.3-bin.zip

to

distributionUrl=https\://services.gradle.org/distributions/gradle-**8.4**-bin.zip

I do get less errors but ./gradlew clean docker still fails:

* What went wrong:
An exception occurred applying plugin request [id: 'com.palantir.docker', version: '0.34.0']
> Failed to apply plugin 'com.palantir.docker'.
   > Could not find matching constructor for: org.gradle.api.internal.artifacts.publish.ArchivePublishArtifact(org.gradle.api.tasks.bundling.Zip_Decorated)

**Comment by erikzwart** [comment]
Editing the plugin in ./build.gradle to
`id 'com.palantir.docker' version '0.35.0'`
Did not resolve issue.
* What went wrong:
A problem occurred evaluating root project 'molgenis-armadillo'.
> Could not get unknown property 'archiveName' for task ':bootJar' of type org.springframework.boot.gradle.tasks.bundling.BootJar.

**Comment by clemens-tolboom** [comment]
I did a stab at it working from the deprecation warning out.

One lingered (forgot which one) so upgraded gradle to 8.0 which failed too.

Comparing with EMX2 weird thing is their `docker` part is same. We use boot jar instead of default.

To test the process we need to run

```sh
./gradlew  --warning-mode all clean build docker assemble
```

to cover most tasks.

### branch(es)

See https://github.com/molgenis/molgenis-service-armadillo/branches/all?query=chore%2Fremo

**Comment by clemens-tolboom** [comment]
Is merged through PR #578
